# EXPLORATORY DATA ANALYSIS (EDA) NETFLIX
   - Data preparation (Cleansing, manipulation, handling  missing value)

Referensi : kaggle.com   

Import libraries that we need to do EDA

In [ ]:
import pandas as pd 
import numpy as np

In [ ]:
# read data netflix.csv
df = pd.read_csv("netflix.csv")

In [ ]:
df # check the top 5 data from nextflix data

show_id     type                             title         director  \
0         s1    Movie              Dick Johnson Is Dead  Kirsten Johnson   
1         s3  TV Show                         Ganglands  Julien Leclercq   
2         s6  TV Show                     Midnight Mass    Mike Flanagan   
3        s14    Movie  Confessions of an Invisible Girl    Bruno Garotti   
4         s8    Movie                           Sankofa     Haile Gerima   
...      ...      ...                               ...              ...   
8785   s8797  TV Show                        Yunus Emre        Not Given   
8786   s8798  TV Show                         Zak Storm        Not Given   
8787   s8801  TV Show                Zindagi Gulzar Hai        Not Given   
8788   s8784  TV Show                              Yoko        Not Given   
8789   s8786  TV Show                               YOM        Not Given   

            country  date_added  release_year rating   duration  \
0     United States   9/25/2021          2020  PG-13     90 min   
1            France   9/24/2021          2021  TV-MA   1 Season   
2     United States   9/24/2021          2021  TV-MA   1 Season   
3            Brazil   9/22/2021          2021  TV-PG     91 min   
4     United States   9/24/2021          1993  TV-MA    125 min   
...             ...         ...           ...    ...        ...   
8785         Turkey   1/17/2017          2016  TV-PG  2 Seasons   
8786  United States   9/13/2018          2016  TV-Y7  3 Seasons   
8787       Pakistan  12/15/2016          2012  TV-PG   1 Season   
8788       Pakistan   6/23/2018          2016   TV-Y   1 Season   
8789       Pakistan    6/7/2018          2016  TV-Y7   1 Season   

                                              listed_in  
0                                         Documentaries  
1     Crime TV Shows, International TV Shows, TV Act...  
2                    TV Dramas, TV Horror, TV Mysteries  
3                    Children & Family Movies, Comedies  
4      Dramas, Independent Movies, International Movies  
...                                                 ...  
8785                  International TV Shows, TV Dramas  
8786                                           Kids' TV  
8787  International TV Shows, Romantic TV Shows, TV ...  
8788                                           Kids' TV  
8789                                           Kids' TV  

[8790 rows x 10 columns]

In [ ]:
# check info data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8790 entries, 0 to 8789
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8790 non-null   object
 1   type          8790 non-null   object
 2   title         8790 non-null   object
 3   director      8790 non-null   object
 4   country       8790 non-null   object
 5   date_added    8790 non-null   object
 6   release_year  8790 non-null   int64 
 7   rating        8790 non-null   object
 8   duration      8790 non-null   object
 9   listed_in     8790 non-null   object
dtypes: int64(1), object(9)
memory usage: 686.8+ KB


- From that info we know, There are 8790 entries data and each column has 8790 entries that means there are no missing value
- In dtype we know about data type of each column. Then we check, is the data type true or not
- After analysis the data type in each column, The data type of data_added is object we should change to date and in column duration we should change the data type to int but before that we should split to 2 categorical : season and minute

In [ ]:
df['date_added']=pd.to_datetime(df['date_added']) # change data type in column data_added to date type

In [ ]:
# # split column duration to 2 categorical : season and minute

# # Making column in minute category
df['duration_minute'] = ""
for i,categori in enumerate(df['duration']):
  if 'min' in categori:
        df.at[i, 'duration_minute'] = categori
  elif 'season' in categori:
        df.at[i, 'duration_minute'] = pd.NA

# Change duration_minute to integer data type
# Make sure you have replaced string 'min' in that column, then you can convert to int
df['duration_minute'] = df['duration_minute'].str.replace(' min','')
df['duration_minute'] = df['duration_minute'].replace('', pd.NA)  # replace empty string with Nan
df['duration_minute'] = df['duration_minute'].astype(pd.Int32Dtype())

In [ ]:
df

show_id     type                             title         director  \
0         s1    Movie              Dick Johnson Is Dead  Kirsten Johnson   
1         s3  TV Show                         Ganglands  Julien Leclercq   
2         s6  TV Show                     Midnight Mass    Mike Flanagan   
3        s14    Movie  Confessions of an Invisible Girl    Bruno Garotti   
4         s8    Movie                           Sankofa     Haile Gerima   
...      ...      ...                               ...              ...   
8785   s8797  TV Show                        Yunus Emre        Not Given   
8786   s8798  TV Show                         Zak Storm        Not Given   
8787   s8801  TV Show                Zindagi Gulzar Hai        Not Given   
8788   s8784  TV Show                              Yoko        Not Given   
8789   s8786  TV Show                               YOM        Not Given   

            country date_added  release_year rating   duration  \
0     United States 2021-09-25          2020  PG-13     90 min   
1            France 2021-09-24          2021  TV-MA   1 Season   
2     United States 2021-09-24          2021  TV-MA   1 Season   
3            Brazil 2021-09-22          2021  TV-PG     91 min   
4     United States 2021-09-24          1993  TV-MA    125 min   
...             ...        ...           ...    ...        ...   
8785         Turkey 2017-01-17          2016  TV-PG  2 Seasons   
8786  United States 2018-09-13          2016  TV-Y7  3 Seasons   
8787       Pakistan 2016-12-15          2012  TV-PG   1 Season   
8788       Pakistan 2018-06-23          2016   TV-Y   1 Season   
8789       Pakistan 2018-06-07          2016  TV-Y7   1 Season   

                                              listed_in  duration_minute  
0                                         Documentaries               90  
1     Crime TV Shows, International TV Shows, TV Act...             <NA>  
2                    TV Dramas, TV Horror, TV Mysteries             <NA>  
3                    Children & Family Movies, Comedies               91  
4      Dramas, Independent Movies, International Movies              125  
...                                                 ...              ...  
8785                  International TV Shows, TV Dramas             <NA>  
8786                                           Kids' TV             <NA>  
8787  International TV Shows, Romantic TV Shows, TV ...             <NA>  
8788                                           Kids' TV             <NA>  
8789                                           Kids' TV             <NA>  

[8790 rows x 11 columns]

In [ ]:
# Making column in season category
#df['duration_season'] = ""
df['duration_season'] = pd.NA

for i,categori in enumerate(df['duration']):
  if 'Season' in categori:
        #df.at[i, 'duration_season'] = categori
        df.at[i, 'duration_season'] = int(categori.replace(' Seasons', '').replace(' Season', ''))
  elif 'min' in categori:
        df.at[i, 'duration_season'] = pd.NA

# Change duration_season to integer data type
df['duration_season'] = df['duration_season'].astype(pd.Int32Dtype())


In [ ]:
df

show_id     type                             title         director  \
0         s1    Movie              Dick Johnson Is Dead  Kirsten Johnson   
1         s3  TV Show                         Ganglands  Julien Leclercq   
2         s6  TV Show                     Midnight Mass    Mike Flanagan   
3        s14    Movie  Confessions of an Invisible Girl    Bruno Garotti   
4         s8    Movie                           Sankofa     Haile Gerima   
...      ...      ...                               ...              ...   
8785   s8797  TV Show                        Yunus Emre        Not Given   
8786   s8798  TV Show                         Zak Storm        Not Given   
8787   s8801  TV Show                Zindagi Gulzar Hai        Not Given   
8788   s8784  TV Show                              Yoko        Not Given   
8789   s8786  TV Show                               YOM        Not Given   

            country date_added  release_year rating   duration  \
0     United States 2021-09-25          2020  PG-13     90 min   
1            France 2021-09-24          2021  TV-MA   1 Season   
2     United States 2021-09-24          2021  TV-MA   1 Season   
3            Brazil 2021-09-22          2021  TV-PG     91 min   
4     United States 2021-09-24          1993  TV-MA    125 min   
...             ...        ...           ...    ...        ...   
8785         Turkey 2017-01-17          2016  TV-PG  2 Seasons   
8786  United States 2018-09-13          2016  TV-Y7  3 Seasons   
8787       Pakistan 2016-12-15          2012  TV-PG   1 Season   
8788       Pakistan 2018-06-23          2016   TV-Y   1 Season   
8789       Pakistan 2018-06-07          2016  TV-Y7   1 Season   

                                              listed_in  duration_minute  \
0                                         Documentaries               90   
1     Crime TV Shows, International TV Shows, TV Act...             <NA>   
2                    TV Dramas, TV Horror, TV Mysteries             <NA>   
3                    Children & Family Movies, Comedies               91   
4      Dramas, Independent Movies, International Movies              125   
...                                                 ...              ...   
8785                  International TV Shows, TV Dramas             <NA>   
8786                                           Kids' TV             <NA>   
8787  International TV Shows, Romantic TV Shows, TV ...             <NA>   
8788                                           Kids' TV             <NA>   
8789                                           Kids' TV             <NA>   

      duration_season  
0                <NA>  
1                   1  
2                   1  
3                <NA>  
4                <NA>  
...               ...  
8785                2  
8786                3  
8787                1  
8788                1  
8789                1  

[8790 rows x 12 columns]

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8790 entries, 0 to 8789
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   show_id          8790 non-null   object        
 1   type             8790 non-null   object        
 2   title            8790 non-null   object        
 3   director         8790 non-null   object        
 4   country          8790 non-null   object        
 5   date_added       8790 non-null   datetime64[ns]
 6   release_year     8790 non-null   int64         
 7   rating           8790 non-null   object        
 8   duration         8790 non-null   object        
 9   listed_in        8790 non-null   object        
 10  duration_minute  6126 non-null   Int32         
 11  duration_season  2664 non-null   Int32         
dtypes: Int32(2), datetime64[ns](1), int64(1), object(8)
memory usage: 772.7+ KB


- After we split duration column to be 2 parts : 'duration_minute' and 'duration_season', then we drop duration column
- In duration_minute and duration_column there are missing value cause the entries are not full 8790 but it's ok cause it's shared to be 2 columns

In [ ]:
# drop column 'duration'
df.drop(["duration"],axis=1, inplace = True)


In [ ]:
# check again
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8790 entries, 0 to 8789
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   show_id          8790 non-null   object        
 1   type             8790 non-null   object        
 2   title            8790 non-null   object        
 3   director         8790 non-null   object        
 4   country          8790 non-null   object        
 5   date_added       8790 non-null   datetime64[ns]
 6   release_year     8790 non-null   int64         
 7   rating           8790 non-null   object        
 8   listed_in        8790 non-null   object        
 9   duration_minute  6126 non-null   Int32         
 10  duration_season  2664 non-null   Int32         
dtypes: Int32(2), datetime64[ns](1), int64(1), object(7)
memory usage: 704.0+ KB


After handle missing value and data type. We check duplicate

In [ ]:
df[df.duplicated(keep=False)] # check duplicate

Empty DataFrame
Columns: [show_id, type, title, director, country, date_added, release_year, rating, listed_in, duration_minute, duration_season]
Index: []

# Export cleaning data to do visualization in tableau

In [ ]:
df.to_excel('Netflix_Clean.xlsx',index=False)